core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -0,0 +1,180 @@
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import hudson.BulkChange;
+import hudson.ExtensionList;
+import hudson.model.User;
+import hudson.model.UserProperty;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.FullControlOnceLoggedInAuthorizationStrategy;
+import hudson.security.HudsonPrivateSecurityRealm;
+import hudson.security.PermissionAdder;
+import hudson.security.SecurityRealm;
+import hudson.security.csrf.DefaultCrumbIssuer;
+import hudson.util.PluginServletFilter;
+import jenkins.model.Jenkins;
+import jenkins.security.s2m.AdminWhitelistRule;
+
+/**
+ * A Jenkins instance used during first-run to provide a limited set of services while
+ * initial installation is in progress
+ */
+public class SetupWizard {
+    /**
+     * The security token parameter name
+     */
+    public static String initialSetupAdminUserName = "initial-setup-admin-user";
+
+    private final Logger LOGGER = Logger.getLogger(SetupWizard.class.getName());
+
+    public SetupWizard(Jenkins j) throws IOException {
+        User admin;
+        // Create an admin user by default with a 
+        // difficult password
+        if(j.getSecurityRealm() == null || j.getSecurityRealm() == SecurityRealm.NO_AUTHENTICATION) { // this seems very fragile
+            BulkChange bc = new BulkChange(j);
+            
+            HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
+            j.setSecurityRealm(securityRealm);
+            String randomUUID = UUID.randomUUID().toString().replace("-", "").toLowerCase(Locale.ENGLISH);
+            admin = securityRealm.createAccount(SetupWizard.initialSetupAdminUserName, randomUUID);
+            admin.addProperty(new SetupWizard.AuthenticationKey(randomUUID));
+
+            AuthorizationStrategy as = Jenkins.getInstance().getAuthorizationStrategy();
+            for (PermissionAdder adder : ExtensionList.lookup(PermissionAdder.class)) {
+                if (adder.add(as, admin, Jenkins.ADMINISTER)) {
+                    return;
+                }
+            }
+            
+            // Lock Jenkins down:
+            FullControlOnceLoggedInAuthorizationStrategy authStrategy = new FullControlOnceLoggedInAuthorizationStrategy();
+            authStrategy.setAllowAnonymousRead(false);
+            j.setAuthorizationStrategy(authStrategy);
+            
+            // Shut down all the ports we can by default:
+            j.setSlaveAgentPort(-1); // -1 to disable
+            
+            // require a crumb issuer
+            j.setCrumbIssuer(new DefaultCrumbIssuer(false));
+            
+            // set master -> slave security:
+            j.getInjector().getInstance(AdminWhitelistRule.class)
+                .setMasterKillSwitch(false);
```

Comment:
Accomplished more simply by making [this](https://github.com/jenkinsci/jenkins/blob/08b6ff33687883d90b22ec83c385713afe3c6adf/core/src/main/java/jenkins/security/s2m/AdminWhitelistRule.java#L94) return `false` by default.